wpilibj/src/main/java/edu/wpi/first/wpilibj/logging/CSVLogFile.java

```diff
@@ -0,0 +1,86 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.wpilibj.logging;
+
+import java.util.Arrays;
+
+/**
+ * A CSVLogFile writes values to a csv file
+ *
+ * <p>For the CSVLogFile to write log informations, you must call
+ * {@link #log} periodically.
+ */
+public class CSVLogFile {
+  private final LogFile m_logFile;
+
+  /**
+   * Instantiate a LogFile passing in its prefix and its column headings.
+   *
+   * <p>If you want the file to be saved in a existing directory, you can add its
+   * path before the file prefix. Exemple : to save the file in a usb stick on
+   * the roborio ("/media/sda1/") : LogFile("/media/sda1/log").
+   *
+   * @param filePrefix The prefix of the LogFile.
+   * @param columnHeadings Titles of CSVLogFile columns.
+   */
+  public <ValueT> CSVLogFile(String filePrefix, ValueT... columnHeadings) {
+    m_logFile = new LogFile(filePrefix, "csv");
+
+    m_logFile.log("\"Timestamp (ms)\",");
```

Comment:
Is it really necessary?? Maybe it's better off with the user choosing when would he want to add a TimeStamp, and at the specific format that he may want.

Reply:
Yes, it's necessary. You always want a timestamp.